application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java

```diff
@@ -133,50 +141,60 @@ public void testTour() throws Exception
         homePage.nextStep();
         assertEquals("Title 3", homePage.getStepTitle());
         assertEquals("Step 3", homePage.getStepDescription());
-        assertFalse(homePage.hasNextStep());
+        assertTrue(homePage.hasNextStep());
         assertTrue(homePage.hasPreviousStep());
-        assertTrue(homePage.hasEndButton());
+        assertFalse(homePage.hasEndButton());
+
+        // Step 4
+        // Use a second page to test the Multipage feature
+        PageWithTour secondPage = PageWithTour.gotoPage("TourCode", "TourClass");
```

Comment:
Could we avoid that by looking at the Tour global target? If the global target is not the current page and if we are not using the Tour (just look at the tour progression to verify that), then do not do anything.
WDYT?